flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -0,0 +1,521 @@
+  }
+
+  @Override
+  public void serverError(boolean isFatal, String message, String stackTraceString) {
+    String exception = composeException(ERROR_TYPE_SERVER, message, stackTraceString);
+    FlutterInitializer.getAnalytics().sendException(exception, isFatal); // test: serverError()
+  }
+
+  @Override
+  public void serverIncompatibleVersion(String s) {
+  }
+
+  @Override
+  public void serverStatus(AnalysisStatus analysisStatus, PubStatus pubStatus) {
+    assert analysisStatus != null;
+    if (!analysisStatus.isAnalyzing()) {
+      @NotNull HashMap<String, Integer> errorCounts = getTotalAnalysisErrorCounts();
+      int errorCount = 0;
+      errorCount += extractCount(errorCounts, AnalysisErrorType.CHECKED_MODE_COMPILE_TIME_ERROR);
+      errorCount += extractCount(errorCounts, AnalysisErrorType.COMPILE_TIME_ERROR);
+      errorCount += extractCount(errorCounts, AnalysisErrorType.SYNTACTIC_ERROR);
+      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, ERRORS, errorCount); // test: serverStatus()
+      int warningCount = 0;
+      warningCount += extractCount(errorCounts, AnalysisErrorType.STATIC_TYPE_WARNING);
+      warningCount += extractCount(errorCounts, AnalysisErrorType.STATIC_WARNING);
+      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, WARNINGS, warningCount); // test: serverStatus()
+      int hintCount = extractCount(errorCounts, AnalysisErrorType.HINT);
+      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, HINTS, hintCount); // test: serverStatus()
+      int lintCount = extractCount(errorCounts, AnalysisErrorType.LINT);
+      FlutterInitializer.getAnalytics().sendEventMetric(DAS_STATUS_EVENT_TYPE, LINTS, lintCount); // test: serverStatus()
+    }
+  }
+
+  private static int extractCount(@NotNull Map<String, Integer> errorCounts, String name) {
+    //noinspection Java8MapApi,ConstantConditions
+    return errorCounts.containsKey(AnalysisErrorType.CHECKED_MODE_COMPILE_TIME_ERROR) ? errorCounts.get(
+      AnalysisErrorType.CHECKED_MODE_COMPILE_TIME_ERROR) : 0;
+  }
+
+  @Override
+  public void computedExistingImports(String file, Map<String, Map<String, Set<String>>> existingImports) {
+  }
+
+  private static void logCompletion(@NotNull String selection, int prefixLength, @NotNull String eventType) {
+    FlutterInitializer.getAnalytics().sendEventMetric(eventType, selection, prefixLength); // test: acceptedCompletion(), lookupCanceled()
+  }
+
+  void logE2ECompletionSuccessMS(long e2eCompletionMS) {
+    FlutterInitializer.getAnalytics().sendTiming(E2E_IJ_COMPLETION_TIME, SUCCESS, e2eCompletionMS); // test: logE2ECompletionSuccessMS()
+  }
+
+  void logE2ECompletionErrorMS(long e2eCompletionMS) {
+    FlutterInitializer.getAnalytics().sendTiming(E2E_IJ_COMPLETION_TIME, FAILURE, e2eCompletionMS); // test: logE2ECompletionErrorMS()
+  }
+
+  private void logAnalysisError(@Nullable AnalysisError error) {
```

Comment:
I'm concerned this is logging too much. If you have 10,000 errors, we are still trying to log 100 events.
What are we trying to achieve by logging the error codes?

Reply:
We need to log far fewer events. Logging analytics needs to not be the reason why users have poor performance or complain our tools use too much of their bandwidth. I would suggest changing all these analytics so a single analytic event or a couple of events are logged summarizing the state of all errors reported rather than ever emitting events proportional to the number of errors or files.

Reply:
I think this will be straight-forward to change. I'll drop the call to `logAnalysisError` for each error and clean up unused elements. We already log the summary info in `serverStatus`.

Reply:
Done! PTAL